src/strategies/meebitsdao/index.ts

```diff
@@ -0,0 +1,70 @@
+import { multicall } from '../../utils';
+import fetch from 'cross-fetch';
+
+export const author = 'peters-josh';
+export const version = '0.1.0';
+
+const abi = [
+  'function ownerOf(uint256 tokenId) public view returns (address owner)',
+  'function tokenURI(uint256 tokenId) public view returns (string uri)'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const uriCalls: any[] = [];
+  for (let i = options.startingTokenId; i <= options.endingTokenId; i++) {
+    uriCalls.push([options.address, 'tokenURI', [i]]);
+  }
+
+  const ownerCalls: any[] = [];
+  for (let i = options.startingTokenId; i <= options.endingTokenId; i++) {
+    ownerCalls.push([options.address, 'ownerOf', [i]]);
+  }
+
+  const uriResponse = await multicall(network, provider, abi, uriCalls, {
+    blockTag
+  });
+
+  const ownerResponse = await multicall(network, provider, abi, ownerCalls, {
+    blockTag
+  });
+
+  async function checkActivated(address) {
+    const index = ownerResponse.findIndex(
+      (res: any) => res.owner.toLowerCase() === address.toLowerCase()
+    );
+    if (index == -1) {
+      return 0;
+    }
+
+    const uri = uriResponse[index][0];
+    let ipfs = uri.split('/');
+    ipfs = ipfs[ipfs.length - 1];
+    const resp = await fetch(`https://${options.gateway}/ipfs/${ipfs}`);
+    const metadata = await resp.json();
+    const activated = metadata.attributes[2];
+    if (activated.trait_type === 'Activated' && activated.value === 'True') {
+      return 1;
+    }
+    return 0;
+  }
+
+  const results = addresses.map(async (address: any) =>
+    checkActivated(address)
```

Comment:
Hi @peters-josh The strategy should not send a request for each voters, this doesn't scale for more addresses . Can you find out any alternative way to get for all addresses in a single call

https://docs.snapshot.org/strategies/create#4-make-sure-you-pass-the-checklist

Reply:
Hi @ChaituVR 

This strategy is dealing with erc721 tokens — every one of which has unique metadata. For every token, have have to get the metadata from the token uri to to check if it contains the specific property we are looking for. I do not believe there is a way around this requirement.

In practice, there are only a maximum of 200 tokens we would ever call, and in reality it will be much less as every token holder will not vote on every issue. The call to fetch the metadata is only made after we confirm the address has a token as well, so this call will not be made on every address.  

What do you suggest as an alternative when creating a strategy that reads ERC721 metadata if we cannot get the metadata for the unique tokens we are dealing with?

Reply:
There will max 200 tokens, but problem is if 200 addresses own it and they vote, Snapshot's server will be sending 200 requests to IPFS every time someone opens the proposal, ofcourse what you said is also makes sense as every token holder will not vote on every issue. but should look at scalability at maximum possibility

Reply:
As addresses in the Discord, this metadata is dynamic so the metadata for every token will need to be checked. It is possible to aggregate the metadata, but this introduces an element of centralization which is not ideal.

Reply:
Do you have that data on a subgraph by chance?

Reply:
@bonustrack @ChaituVR I updated the strategy so it no longer pulls the metadata for each address (uses one centralized API at the beginning instead).